redwood-gradle-plugin/src/main/kotlin/app/cash/redwood/gradle/RedwoodComposeExtension.kt

```diff
@@ -39,7 +36,35 @@ public abstract class RedwoodComposeExtension
    * }
    * ```
    */
-  public val kotlinCompilerPlugin: Property<String> =
-    objectFactory.property(String::class.java)
-      .convention(composeCompilerVersion)
+  public val kotlinCompilerPlugin: Property<String>
+
+  /**
+   * Enable the output of metrics from the Compose compiler.
+   *
+   * Text files will be written to `generated/redwood/compose-metrics/` in the project's build
+   * directory. See
+   * [the compiler documentation](https://github.com/androidx/androidx/blob/androidx-main/compose/compiler/design/compiler-metrics.md#reports-breakdown)
+   * for more information about the contents.
+   *
+   * **NOTE:** This should only be enabled during investigation as it breaks the use of
+   * Gradle's build cache for this project's Kotlin compilation tasks.
+   *
+   * @see enableReports
+   */
+  public fun enableMetrics()
```

Comment:
Worth incorporating the scary consequence into the function name?

```
enableMetrics_ThisBreaksGradleBuildCaching_()
```

I might inadvertently check in code that calls enableMetrics(), but my code reviewer would catch that!

Reply:
It wouldn't actually be the end of the world if you did so. So maybe we can find something that's still cautionary but not as in-your-face. Will sleep on it.

Reply:
What about wrapping them into something else like
```groovy
redwood {
  dangerZone {
    enableMetrics()
  }
}
```
Is that enough to signal you should learn more about why it's dangerous?

Reply:
Done. Also added tests.